src/HospitalLibrary/Auth/AuthService.cs

```diff
@@ -51,7 +51,8 @@ public async Task<List<Allergen>> GetPatientsAllergens(List<AllergenDTO> allerge
         }
         public async Task<Doctor> GetPatientsDoctor(string doctorUid)
         {
-            foreach (Doctor doctor in await _unitOfWork.DoctorRepository.GetTwoUnburdenedDoctors())
+            Doctor mostUnburdened = await _unitOfWork.DoctorRepository.GetMostUnburdenedDoctor();
+            foreach (Doctor doctor in await _unitOfWork.DoctorRepository.GetUnburdenedDoctors(mostUnburdened))
```

Comment:
You dont have to pass whole Doctor to function GetUnburdenedDoctors, you can pass just mostUnburdened.Patients.Count